### PR TITLE
test: first tests, and coverage reported to Codacy from CI

### PR DESCRIPTION
**Pull request by valinmalach**
## What

The repo had no tests and no CI, so Codacy had no coverage to show. This adds a starting suite and a workflow that uploads its coverage.

- **Tests** on the two modules whose rules are pure: `services/birthday.py` (`next_birthday_on`, `next_birthday`) and `services/twitch/migrate_plan.py` (`decide`, `condition_of`). 12 tests; nothing touches Discord, Twitch or Postgres.
- **`.github/workflows/coverage.yml`** runs `pytest --cov` on every push and uploads `coverage.xml` with `codacy/codacy-coverage-reporter-action`. The upload is skipped when `CODACY_PROJECT_TOKEN` is empty (forks, Dependabot), so a missing secret cannot fail passing tests.
- **Config**: pytest and coverage in `pyproject.toml`; asserts allowed under `tests/` in Ruff, Bandit and `.codacy.yaml`.
- **Docs**: `AGENTS.md`, `VERITY.md` and the Standard's rationale said there was no suite or CI; they now say what exists. The coverage threshold stays 0.

Four commits, dependency bump first (`uv.lock` on its own).

## Worth knowing

- Coverage is **14%**, and that is the honest figure. It first read 26% because coverage.py drops files in directories without an `__init__.py` (`cogs/`, `services/twitch/`) from the total instead of counting them at 0%. `include_namespace_packages = true` fixes it, and it belongs under `[tool.coverage.report]`; under `[run]` it only warns "Unrecognized option" and changes nothing.
- The workflow triggers on `push`, not `pull_request`: Codacy needs coverage for a PR's head commit and the commit it branched from, and a `pull_request` run reports a merge commit that neither is.
- `master` has no coverage yet, so this PR has no baseline to compare against until it merges.
- `astral-sh/setup-uv` publishes no floating major tag, so its patch (`v10.1.0`) is pinned.

## Verification

Sourcery, `ruff format`, `ruff check` and `pyright` are clean, and the 12 tests pass locally. The workflow passes `actionlint` but had not run on GitHub when this was written. Verity gave WARN with 0 code issues on each commit; its intent note described an unrelated task and does not apply.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Establish an initial pure-logic test suite and publish its coverage from CI to Codacy.

New Features:
- Add initial pytest coverage for birthday calculations and Twitch subscription migration rules.

Enhancements:
- Configure pytest and coverage reporting, including accurate coverage totals for namespace packages.
- Allow test assertions and generated dependency lock data in static-analysis configurations.
- Document the initial test suite, its scope, and current coverage expectations.

Build:
- Add pytest and pytest-cov development dependencies and lockfile updates.

CI:
- Run tests with coverage on every push and upload coverage reports to Codacy when credentials are available.

Tests:
- Add focused tests for birthday edge cases and Twitch migration decisions and conditions.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+"""
+
+import os
+
+os.environ |= {
+    "DISCORD_TOKEN": "test",
+    "TWITCH_CLIENT_ID": "test",
+    "TWITCH_CLIENT_SECRET": "test",
+    "TWITCH_WEBHOOK_SECRET": "test",
+    "DATABASE_URL": "postgresql://test:test@localhost:5432/test",
```
[the remark this answers is not in the document]
Comment (valinmalach) [reply tests/conftest.py]: Not changed. The entry is well-formed: line 15 of `tests/conftest.py` is `"DATABASE_URL": "postgresql://test:test@localhost:5432/test",`, with its closing quote and trailing comma. Checked at the commented commit (`0e22289`): the file parses with `ast.parse`, all 12 tests pass, and the Coverage workflow is green on it. A syntax error here would have failed test collection. The `//` inside the URL string is the likely thing that was misread as a comment.
